app/javascript/flavours/glitch/features/composer/index.js

```diff
@@ -227,8 +246,15 @@ const handlers = {
       return;
     }
 
-    //  Submits the status.
-    if (onSubmit) {
+    // Submit unless there are media with missing descriptions
+    if (mediaDescriptionConfirmation && onMediaDescriptionConfirm && media && media.some(item => !item.get('description'))) {
+      const firstWithoutDescription = media.findIndex(item => !item.get('description'));
+      const inputs = document.querySelectorAll('.composer--upload_form--item input');
```

Comment:
You could keep a reference to the rendered component and only do `.querySelectorAll` in there.